packages/core/src/api/lookup.ts

```diff
@@ -1154,7 +1154,8 @@ function textContainsTokens(text: string, tokens: QueryTokens) {
   const lowerCasedText = text.toLowerCase();
 
   const createTagPattern = (token: string) => {
-    return `<${MATCH_TAG_NAME}\\s+id="(.+?)">${token}<\\/${MATCH_TAG_NAME}>`;
+    const escapedToken = token.replace(/[()]/g, "\\$&");
+    return `<${MATCH_TAG_NAME}\\s+id="(.+?)">${escapedToken}<\\/${MATCH_TAG_NAME}>`;
```

Comment:
We are already escaping tokens in `query-transformer`. Why is that not working?

Reply:
I found this in `query-transformer.ts` https://github.com/streetwriters/notesnook/blob/master/packages/core/src/utils/query-transformer.ts#L133

I think we need to escape the `(` or `)` characters by doing this `.replace(/[()]/g, "\\$&")` to prevent regex errors

Reply:
Looking at it again, I think your change is more appropriate.